### PR TITLE
feat: Navigate to node from search results

### PR DESCRIPTION
**Pull request by DafyddLlyr**
## What does this PR do?
 - Uses [the planx-core function `getPathForNode()`](https://github.com/theopensystemslab/planx-core/pull/506/) to construct a PlanX relative URL
 - Wires this up to search result cards

Test flow on Pizza - https://3650.planx.pizza/testing/flow-navigation-search

Prior art - https://github.com/theopensystemslab/planx-new/pull/3460

https://github.com/user-attachments/assets/e10a21c5-1614-4358-a57c-1e4d42786019

## To Do (this PR)
- Bump planx-core across all projects once https://github.com/theopensystemslab/planx-core/pull/506/ approved and merged

## Next step...
- Portal "wrapper" UI for cards

### CODE REVIEW

File: editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts [hunk 1]
```diff
@@ -487,4 +492,29 @@ export const editorStore: StateCreator<
...
+    // Construct the internal portal path if applicable
+    const portalPath = internalPortals.length
+      ? "," + internalPortals.map(({ id }) => id).join(",")
+      : "";
+
+    // Determine node path based on the node type
+    const nodePath =
+      node.type === ComponentType.Answer
+        ? `nodes/${grandparent.id}/nodes/${parent.id}/edit`
+        : `nodes/${parent.id}/nodes/${node.id}/edit`;
```
Comment (DafyddLlyr) [inline editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts:515]: I _think_ this is the only variance that controls the path used for the modal - please let me know if there's something I've missed here!

Comment (jessicamcinchak) [reply editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts:515]: This seems right to me here! Only edge case I possibly considered here were "Filters" - but these links work as expected testing on pizza :raised_hands: 

Plus very cool fun bonus - Search now makes visible the (wild!) fact that Filter's have a default data field `flag` assigned to them via the component's `handleSubmit` and later relied on in `upcomingCardIds()`: 
Screenshot from 2024-09-11 21-35-30

File: editor.planx.uk/src/pages/FlowEditor/lib/__tests__/getURLForNode.test.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    expect(url).toEqual(
+      "/testTeam/testFlow,tarantinoPortal,moviesPortal/nodes/tarantinoPortal/nodes/tarantinoMovieQuestion/edit",
+    );
+  });
+
+  test("an answer node, deeply within internal portals", () => {
+    const url = getURLForNode("reservoirDogsAnswer");
+    expect(url).toEqual(
+      "/testTeam/testFlow,tarantinoPortal,moviesPortal/nodes/tarantinoPortal/nodes/tarantinoMovieQuestion/edit",
+    );
```
Comment (jessicamcinchak) [inline editor.planx.uk/src/pages/FlowEditor/lib/__tests__/getURLForNode.test.ts:65]: :muscle:

**Review by jessicamcinchak — APPROVED** [review]
Very good work - can already think of so many other places I want to use `getURLForNode` (publish changelogs, validation checks, etc!!) :tada:

File: editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts (hunk 1 above)
Comment (DafyddLlyr) [reply editor.planx.uk/src/pages/FlowEditor/lib/store/editor.ts:515]: Ha! Good catch and a fun little easter egg... 🙃

**Pull request merged**